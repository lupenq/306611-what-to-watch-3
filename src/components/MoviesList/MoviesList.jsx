import SmallMovieCard from '../SmallMovieCard/SmallMovieCard';

class MoviesList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {activeHoverMovieName: ``};

    this.movieCardHoverHandler = this.movieCardHoverHandler.bind(this);
    this.movieCardUnhoverHandler = this.movieCardUnhoverHandler.bind(this);
  }

  movieCardHoverHandler(name) {
    this.setState({activeHoverMovieName: name});
  }

  movieCardUnhoverHandler() {
    this.setState({activeHoverMovieName: ``});
  }

  render() {
    const {filmsList} = this.props;

    return <div className="catalog__movies-list">
      {
        filmsList.map((item, index) =>(
          <SmallMovieCard
            name={item.name}
            picture={item.picture}
            key={item.name + index}
            onMovieCardHover={this.movieCardHoverHandler}
            onMovieCardUnhover={this.movieCardUnhoverHandler}
          />
        )
        )
      }
    </div>;
  }
}

MoviesList.propTypes = {
  filmsList: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        picture: PropTypes.string.isRequired
      })
  ),
  onMovieCardTitleClick: PropTypes.func.isRequired
};

export default MoviesList;
