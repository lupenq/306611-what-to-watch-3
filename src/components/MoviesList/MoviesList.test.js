import MoviesList from './MoviesList';

it(`Render MoviesList`, () => {
  const IMG_URL = `https://api.adorable.io/avatars/128`;

  const films = [
    {
      name: `Интерстеллар`,
      picture: `${IMG_URL}`
    },
    {
      name: `Ирландец`,
      picture: `${IMG_URL}`
    },
    {
      name: `Один дома`,
      picture: `${IMG_URL}`
    },
    {
      name: `Танго в Париже`,
      picture: `${IMG_URL}`
    },
    {
      name: `Зелёная книга`,
      picture: `${IMG_URL}`
    },
    {
      name: `Мимино`,
      picture: `${IMG_URL}`
    },
    {
      name: `Человек-паук`,
      picture: `${IMG_URL}˙}`
    },
    {
      name: `Бойцовский клуб`,
      picture: `${IMG_URL}}`
    }
  ];


  const tree = renderer
  .create(
      <MoviesList
        filmsList={films}
        onMovieCardTitleClick={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
