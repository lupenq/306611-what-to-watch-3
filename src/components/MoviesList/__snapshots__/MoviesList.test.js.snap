// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MoviesList 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Интерстеллар"
        height="175"
        src="https://api.adorable.io/avatars/128"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Интерстеллар
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Ирландец"
        height="175"
        src="https://api.adorable.io/avatars/128"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Ирландец
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Один дома"
        height="175"
        src="https://api.adorable.io/avatars/128"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Один дома
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Танго в Париже"
        height="175"
        src="https://api.adorable.io/avatars/128"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Танго в Париже
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Зелёная книга"
        height="175"
        src="https://api.adorable.io/avatars/128"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Зелёная книга
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Мимино"
        height="175"
        src="https://api.adorable.io/avatars/128"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Мимино
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Человек-паук"
        height="175"
        src="https://api.adorable.io/avatars/128˙}"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Человек-паук
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Бойцовский клуб"
        height="175"
        src="https://api.adorable.io/avatars/128}"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Бойцовский клуб
      </a>
    </h3>
  </article>
</div>
`;
